### PR TITLE
add mailgun docs

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

File: appengine/mail/mailgun/mailgun.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
+Sending email with Mailgun
+=================================
+
+Overview
+--------
+
+The Mailgun API is built on HTTP. Our API is RESTful_ and it:
```
Comment (tmatsuo) [inline appengine/mail/mailgun/mailgun.rst]: "Our API" sounds not right.

File: appengine/mail/mailgun/mailgun.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * Include the Autoloader
+ * Run the following for the latest version:
+ *
+ *   $ composer require mailgun/mailgun-php
+ *
+ */
+require 'vendor/autoload.php';
```
Comment (tmatsuo) [inline appengine/mail/mailgun/mailgun.php]: Test?

File: appengine/mail/mailgun/mailgun.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
```
Comment (tmatsuo) [inline appengine/mail/mailgun/mailgun.php]: license header

**Comment by bshaffer** [comment]
@tmatsuo rewritten as a silex app to mirror [python's approach](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/appengine/mailgun/app.yaml)

**Comment by bshaffer** [comment]
blocked by https://github.com/mailgun/mailgun-php/pull/94

**Comment by bshaffer** [comment]
closing in favor of #42 so we can run tests

**Pull request closed without merging**